ras_backstage/resources/sample/post_sample_file.py

```diff
@@ -0,0 +1,38 @@
+import logging
+
+from flask import jsonify, make_response, request
+from flask_restplus import Resource, reqparse
+from structlog import wrap_logger
+from werkzeug.datastructures import FileStorage
+
+from ras_backstage import sample_api
+from ras_backstage.controllers import sample_controller
+
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+parser = reqparse.RequestParser()
+parser.add_argument('file', location='files', type=FileStorage, required=True)
+parser.add_argument('collection_exercise_id', location='args', required=True)
+
+
+@sample_api.route('/upload')
+class PostSampleFile(Resource):
+
+    @staticmethod
+    @sample_api.expect(parser)
+    def post():
+        collection_exercise_id = request.args['collection_exercise_id']
```

Comment:
The parser doesn't actually enforce the `required=True` for query string arguments so if you try and post without `?collection_exercise_id=asd....` in the url it will fail on this line.

You do still get a 400 response as is right now, the error message just wont be that helpful

Would this endpoint work better with endpoint parameters? i.e. `/upload/<collection_exercise_id>`

Reply:
If it doesn't enforce them then it's probably worth using a parameter instead. 👍